### PR TITLE
Update FTS aerotech agent to work with multiple different FTSes

### PR DESCRIPTION
**Pull request by tdalford1**
## Description
Updates fts_aerotech_agent to work with an inputted .YAML config file stored in $OCS_CONFIG_DIR. This config includes some FTS-specific parameters which vary depending on the FTS used or the type of data being taken.

## Motivation and Context

Originally the FTS params were hard-coded in before the class definition. Making the params configurable from a YAML file creates flexibility if they need to change and is necessary since we now have two different FTSes in use.

## How Has This Been Tested?

Awhile back I tested the changes in the LATRt test environment and I was able to work both Uchicago's original FTS and the one we were sending to UCSD with this branch. Admittedly I haven't tested this in awhile though and it could be that things are broken with updated socs. And I haven't used any official test suites.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

File: agents/fts_aerotech_stage/fts_aerotech_agent.py [hunk 1]
```diff
@@ -286,6 +308,7 @@ def make_parser(parser=None):
     pgroup = parser.add_argument_group('Agent Options')
     pgroup.add_argument('--ip-address')
     pgroup.add_argument('--port')
+    pgroup.add_argument('--config_file')
```
Comment (BrianJKoopman) [inline agents/fts_aerotech_stage/fts_aerotech_agent.py:311]: While there is a mix of '-' and '_' separators in multi-word flags, '-' is more common among our Agents. Can we change this to `--config-file`? (Note, `args.config_file` will still need to be used on line 335.)

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
These changes looks good. A couple of small change requests inline, mostly with documentation.

File: agents/fts_aerotech_stage/fts_aerotech_agent.py [hunk 2]
```diff
@@ -107,10 +104,10 @@ class FTSAerotechAgent:
         port: Port of Motion Controller
         mode: 'acq': Start data acquisition on initialize
         samp: default sampling frequency in Hz
- 
+
     """
 
-    def __init__(self, agent, ip_addr, port, mode=None, samp=2):
+    def __init__(self, agent, ip_addr, port, config_file, mode=None, samp=2):
```
[the remark this answers is not in the document]
Comment (tdalford1) [reply agents/fts_aerotech_stage/fts_aerotech_agent.py:118]: done.

File: agents/fts_aerotech_stage/fts_aerotech_agent.py [hunk 3]
```diff
@@ -129,15 +126,39 @@ def __init__(self, agent, ip_addr, port, mode=None, samp=2):
...
-                                 buffer_time = 0)
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+        # Load dictionary of specific mirror paramters, since some parameters
+        # like limits and translate vary over different FTSes This is loaded
+        # from a yaml file, which is assumed to be in the $OCS_CONFIG_DIR
+        # directory.
+        if config_file == 'None':
```
[the remark this answers is not in the document]
Comment (tdalford1) [reply agents/fts_aerotech_stage/fts_aerotech_agent.py]: sorry, I think just pointed grace to this as an example awhile ago. I agree with this None change and have changed it accordingly

File: docs/agents/fts_agent.rst [hunk 4]
```diff
@@ -31,10 +31,22 @@ the available arguments::
         'arguments': [
           ['--ip-address', '192.168.10.13'],
           ['--port', 8000],
+	  ['--config_file', 'fts_config.yaml'],
```
[the remark this answers is not in the document]
Comment (tdalford1) [reply docs/agents/fts_agent.rst]: sorry, changed. I also slightly modified the indentation of the block after this too to match the next time

File: agents/fts_aerotech_stage/fts_aerotech_agent.py (hunk 1 above)
Comment (tdalford1) [reply agents/fts_aerotech_stage/fts_aerotech_agent.py]: sounds good, changed.

**Comment by tdalford1** [comment]
Hi, sorry for the delay, I have tried to update the branch to include the suggestions you mentioned. Should I do anything currently about the merge conflicts with the develop branch?

**Pull request merged**